ghcjs-src/Miso/Subscription/WebSocket.hs

```diff
@@ -151,48 +129,207 @@ data SocketState
   | CLOSED     -- ^ 3
   deriving (Show, Eq, Ord, Enum)
 
--- | Retrieves current status of `WebSocket`
-getSocketState :: IO SocketState
-getSocketState = do
-  Just ws <- readIORef websocket
-  toEnum <$> getSocketState' ws
+-- | Low-level websocket handle, wraps the raw JSVal.
+newtype WebSocket_ = WebSocket_ JSVal
```

Comment:
Maybe call this `JSWebSocket` to make it easier to infer what this represents based on the name.

Reply:
Sounds good